Clarify configs in self-hosted Event Handler guides

ptgott opened the pull request:
Closes #38184

Prevent users from using example Event Handler configurations without modifying the placeholder values. Remove placeholder values from example Event Handler configs and add them to the comments accompanying the modified fields.

Exceptions are if the accompanying guide does not explain the placeholder, e.g., `serviceType` and the `log` section. In these cases, we expect users to copy the placeholders and use them as they are.

Skip purely hosted Event Handler guides, which don't include instructions for configuring the Event Handler.

Note that in the MS Teams guide, the TOML example includes clear placeholders to indicate format, and this change leaves these in.

Also note that some guides, like the PagerDuty guide, include an example of the final config files after providing lists of instructions. This change does not touch the example configs in this case.

ptgott committed:
Edit self-hosted Event Handler guides

Closes #38184

Prevent users from using example Event Handler configurations without
modifying the placeholder values. Remove placeholder values from example
Event Handler configs and add them to the comments accompanying the
modified fields.

Exceptions are if the accompanying guide does not explain the
placeholder, e.g., `serviceType` and the `log` section. In these cases,
we expect users to copy the placeholders and use them as they are.

Skip purely hosted Event Handler guides, which don't include
instructions for configuring the Event Handler.

Note that in the MS Teams guide, the TOML example includes clear
placeholders to indicate format, and this change leaves these in.

Also note that some guides, like the PagerDuty guide, include an example
of the final config files after providing lists of instructions. This
change does not touch the example configs in this case.

kopiczko commented on examples/resources/plugins/teleport-jira-cloud.toml:
```diff
@@ -1,24 +1,24 @@
 # Example Jira plugin configuration TOML file
 [teleport]
 # Proxy Service domain and HTTPS port
-auth_server = "myinstance.teleport.sh:443"
+auth_server = ""
```
No example here?

kopiczko commented on examples/resources/plugins/teleport-email-helm.yaml:
```diff
@@ -1,14 +1,14 @@
-  address: teleport.example.com:443
-  identitySecretName: teleport-plugin-email-identity
-  identitySecretPath: identity
+  address: ""              # e.g., teleport.example.com:443 
```
Could be replaced with `*.teleport.sh` domain

kopiczko commented on examples/resources/plugins/teleport-jira-cloud.toml:
```diff
@@ -1,24 +1,24 @@
 # URL on which webhook server is accessible externally, for example,
 # [https://]teleport-jira.example.com
-public_addr = "example.com" 
-https_key_file = "/var/lib/teleport/plugins/jira/server.key"  # TLS private key
-https_cert_file = "/var/lib/teleport/plugins/jira/server.crt" # TLS certificate
+public_addr = "" 
+https_key_file = ""  # TLS private key
+https_cert_file = "" # TLS certificate
```
It's a bit confusing. I think all comments for a section should go either above or to the side.

kopiczko commented on examples/resources/plugins/teleport-msteams.toml:
```diff
@@ -24,8 +24,8 @@ preload = true
-addr = "localhost:3025"
-identity = "identity"
+addr = "" # e.g., localhost:3025
+identity = ""
```
This identity looks a bit sad, no example, feels like it could be left empty.

ptgott committed:
Respond to kopiczko feedback

- Fill in missing examples.
- Consistently place comments above configuration fields.
- Use Cloud domain in the email Helm example.
